src/scripts/DataLayer.js

```diff
@@ -313,21 +313,41 @@ DataLayer.Manager.prototype._isMatching = function(listener, item) {
 
   if (item.type === DataLayer.constants.itemType.DATA) {
     if (listenerConfig.on === DataLayer.constants.event.CHANGE) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
   } else if (item.type === DataLayer.constants.itemType.EVENT) {
     if (listenerConfig.on === DataLayer.constants.event.EVENT ||
       listenerConfig.on === itemConfig.event) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
     if (itemConfig.data &&
       listenerConfig.on === DataLayer.constants.event.CHANGE) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
   }
   return isMatching;
 };
 
+/**
+ * Checks if the given listerConfig has a selector and if so, is it matching with the data payload of the itemConfig.
```

Comment:
I'd rephrase it e.g.:
Checks if the given listenerConfig has a selector that points to an object in the data payload of the itemConfig.